src/thread_task.rs

```diff
@@ -301,68 +90,60 @@ impl Rav1dTasks {
     fn merge_pending_frame(&self, c: &Rav1dContext) -> bool {
         let merge = self.pending_tasks_merge.swap(false, Ordering::SeqCst);
         if merge {
-            let mut pending_tasks = self.pending_tasks.lock();
-            let range = {
-                let mut tasks = self.tasks.try_write().unwrap();
-                if self.head.load(Ordering::Relaxed).is_none() {
-                    tasks.clear();
+            let mut tasks = self.tasks.try_write().unwrap();
+            let new_start = tasks.len();
+
+            {
+                let mut pending_tasks = self.pending_tasks.lock();
+                if pending_tasks.len() == 0 {
+                    // This is a 'spurious' merge: that is, `merge` is set but the vector is empty.
```

Comment:
Is the optimization from skipping the actual merge (`.extend` and `.drain`, or more efficiently, `.append), or from returning `false`?